lvsfunc/misc.py

```diff
@@ -295,35 +287,41 @@ def _frames_since_bookmark(n: int, clip: vs.VideoNode, bookmarks: List[int]) ->
     return core.std.FrameEval(clip, partial(_frames_since_bookmark, clip=clip, bookmarks=bookmarks))
 
 
-def allow_vres(func: Callable[..., vs.VideoNode],
-               format_out: Optional[int] = None) -> Callable[..., vs.VideoNode]:
+def allow_variable(width: Optional[int] = None, height: Optional[int] = None,
+                   format: Optional[int] = None
+                   ) -> Callable[..., Callable[..., vs.VideoNode]]:
```

Comment:
with a Typevar, this would then return `Callable[[F], F]`.

Reply:
mypy doesn't seem to like this and thinks that any calls from decorated functions will return `Any`

Reply:
Sad but can't be helped.